src/cpu-template-helper/src/strip/x86_64.rs

```diff
@@ -0,0 +1,418 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashSet;
+
+use vmm::guest_config::templates::CustomCpuTemplate;
+
+use crate::strip::remove_common;
+
+#[allow(dead_code)]
+pub fn strip(templates: Vec<CustomCpuTemplate>) -> Vec<CustomCpuTemplate> {
+    // Convert `Vec<CustomCpuTemplate>` to two `Vec<HashSet<_>>` of modifiers.
+    let (mut cpuid_modifiers_sets, mut msr_modifiers_sets): (Vec<_>, Vec<_>) = templates
+        .into_iter()
+        .map(|template| {
+            (
+                template.cpuid_modifiers.into_iter().collect::<HashSet<_>>(),
+                template.msr_modifiers.into_iter().collect::<HashSet<_>>(),
+            )
+        })
+        .unzip();
+
+    // Remove common items.
+    remove_common(&mut cpuid_modifiers_sets);
+    remove_common(&mut msr_modifiers_sets);
+
+    // Convert back to `Vec<CustomCpuTemplate>`.
+    cpuid_modifiers_sets
+        .into_iter()
+        .zip(msr_modifiers_sets.into_iter())
+        .map(|(cpuid_modifiers_set, msr_modifiers_set)| {
+            let mut cpuid_modifiers = cpuid_modifiers_set.into_iter().collect::<Vec<_>>();
+            let mut msr_modifiers = msr_modifiers_set.into_iter().collect::<Vec<_>>();
+
+            cpuid_modifiers.sort_by_key(|modifier| (modifier.leaf, modifier.subleaf));
+            msr_modifiers.sort_by_key(|modifier| modifier.addr);
+
+            CustomCpuTemplate {
+                cpuid_modifiers,
+                msr_modifiers,
+            }
+        })
+        .collect::<Vec<_>>()
+}
+
+#[cfg(test)]
+mod tests {
+    use vmm::guest_config::templates::x86_64::{
+        CpuidLeafModifier, CpuidRegister, CpuidRegisterModifier, RegisterModifier,
+        RegisterValueFilter,
+    };
+
+    use super::*;
+
+    // Summary of CPUID modifiers:
+    // * As CPUID leaf 0x0 / subleaf 0x0 modifier exists in all the templates and its values are
+    //   different, it should be removed.
+    // * As CPUID leaf 0x1 / subleaf 0x0 modifier only exists in the second template, it should be
+    //   preserved.
+    // * As CPUID leaf 0x2 / subleaf 0x1 modifier exists in all the templates but EBX values are
+    //   different, the leaf modifier (not only the register modifier EBX) should be preserved.
+    fn build_input_cpuid_templates() -> Vec<CustomCpuTemplate> {
+        vec![
+            CustomCpuTemplate {
```

Comment:
Nit: I'm wondering if there is a way to use a macro to reduce verbosity of the tests. Something like:
```rust
CustomCpuTemplate {
  cpuid_modifiers: vec![
    cpuid_mod!(0x0, 0x0, [Eax], [0x0]);
    cpuid_mod!(0x0, 0x0, [Eax, Ebx], [0x0, 0x0]);
  ],
  msr_modifiers: vec![],
}
```
I admit that the macro itself may look too complex. Please feel to ignore in that case.

Reply:
I think it's worth doing! Let me try...

Reply:
Done. It's much easier to see! Thanks for great suggestion!